README.md

```diff
@@ -60,6 +61,35 @@ which contain a toctree need to have the suffix `.trst` instead of just `.rst`.
 This way the projects which use this as a submodule are able to ignore the
 toctrees and build their own instead.
 
+### Understanding Spell Check
+A spell check is performed during the build step by default. It uses in-built
+language specific dictionaries and project specific dictionaries (aspell.en.pws)
+to verify the spellings and causes the build to fail in case of any typos.
+
+The project, which uses this blueprint, should have its own custom dictionary,
+similar to the one in blueprint (aspell.en.pws). A link to the check_spelling.sh
+in the repository can be created and the spell checker should then run correctly.
+
+To build without spell check, configure cmake as below:
+
+    cmake -H. -Bbuild -DPERFORM_SPELL_CHECK=OFF
+
+To invoke the spell check utility directly, run the command below:
+
+    ./check_spelling.sh
+
+The spell checker has various modes and options, which can be seen by:
+
+    ./check_spelling.sh -usage
+
+New words can be added to the dictionary by the script interface itself. Run
+the script as below to view all typos and add one or all words:
+
+    ./check_spelling.sh -a
+
+The dictionary can also be modified manually, by adding words to the end of the
```

Comment:
Please add a note on increasing the word count (top of the aspell dict file) when adding new words manually as well. I am not sure what the significance is of the word count, but aspell seems to maintain that number when aspell itself adds words.

Reply:
This is not really needed; The word count is more like a hint.

http://aspell.net/man-html/Format-of-the-Personal-and-Replacement-Dictionaries.html#Format-of-the-Personal-and-Replacement-Dictionaries

Reply:
Ah! Good info. Thanks :)